spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/TaskAppNode.java

```diff
@@ -16,22 +16,34 @@
 
 package org.springframework.cloud.dataflow.core.dsl;
 
+import java.util.Collections;
+import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.Map;
 
 /**
- * Common AST base class for nodes representing job definitions or job references.
+ * Represents either a task application or task definition reference. Task application references
```

Comment:
Given that the parser can be used for both the 'initial dsl':

task create --name "someTask" --definition "fooApp && barApp --p1=v1"

and the 'internal dsl' --graph param:

_someTask.fooApp && _someTask.barApp

Then it does apply to both, representing either a task app or a task def reference.